packages/nix/packages/pf-eks-resume.sh

```diff
@@ -0,0 +1,286 @@
+#!/usr/bin/env bash
+
+# This script restores the environments EKS cluster nodes and resources to their original state
+
+set -eo pipefail
+export AWS_PAGER=""
+
+####################################################################
+# Step 1: Select the cluster
+####################################################################
+
+get_cluster_tag() {
+  local tag_key="$1"
+
+  # Retrieve the tags for the EKS cluster
+  local tag_value
+
+  tag_value=$(aws eks describe-cluster --name "$EKS_CLUSTER_NAME" --query "cluster.tags.\"$tag_key\"" --output text)
+
+  if [ "$tag_value" == "None" ]; then
+    echo "Tag '$tag_key' not found for cluster '$EKS_CLUSTER_NAME'." >&2
+    return 1
+  fi
+
+  # Check if the tag value is "true"
+  if [ "$tag_value" == "true" ]; then
+    echo "Tag '$tag_key' is set to true." >&2
+    return 0
+  else
+    echo "Tag '$tag_key' is not set to true." >&2
+    return 1
+  fi
+}
+
+select_cluster() {
+  EKS_CLUSTER_NAME=$(kubectl config get-contexts -o name | fzf --prompt="Select a Kubernetes cluster: ")
+  CONTEXT_ARGS="--context=$EKS_CLUSTER_NAME"
+
+  aws_profile=$(pf-get-aws-profile-for-kube-context "$EKS_CLUSTER_NAME")
+
+  # check that provided profile is the root user
+  if [[ ! $(aws sts get-caller-identity --profile "$aws_profile" --no-cli-page --output json | jq -r '.Arn') =~ "root" ]]; then
+    echo "Error: Provided profile is not the root user. Exiting." >&2
+    exit 1
+  fi
+
+  export AWS_PROFILE=$aws_profile
+
+  if ! get_cluster_tag "panfactum.com/suspended"; then
+    echo "Cluster '$EKS_CLUSTER_NAME' is not paused. Exiting." >&2
+    exit 1
+  fi
+}
+
+set_tag_to_cluster() {
+  local tag_key=$1
+  local tag_value=$2
+
+  echo "Adding tag '$tag_key=$tag_value' to the EKS cluster '$EKS_CLUSTER_NAME'..." >&2
+
+  if aws eks tag-resource \
+    --resource-arn "arn:aws:eks:$(aws configure get region):$(aws sts get-caller-identity --query Account --output text):cluster/$EKS_CLUSTER_NAME" \
+    --tags "$tag_key=$tag_value"; then
+    echo "Successfully added tag '$tag_key=$tag_value' to the EKS cluster '$EKS_CLUSTER_NAME'." >&2
+  else
+    echo "Failed to add tag to the EKS cluster '$EKS_CLUSTER_NAME'." >&2
+  fi
+}
+
+####################################################################
+# Step 2: Confirmation
+####################################################################
+
+confirm_selection() {
+  echo -e "You selected \n" >&2
+  echo -e "Cluster: $EKS_CLUSTER_NAME\n" >&2
+  read -rp "Enter name of cluster to confirm: " CONFIRM_CLUSTER
+
+  if [ "$EKS_CLUSTER_NAME" != "$CONFIRM_CLUSTER" ]; then
+    echo -e "$CONFIRM_CLUSTER does not match $EKS_CLUSTER_NAME Exiting.\n" >&2
+    exit 1
+  fi
+}
+
+scale_asg_nat_nodes() {
+  local prefix="nat-"
+  local min_size=1
+  local desired_capacity=1
+
+  # Step 1: Get the Subnet IDs associated with the EKS cluster
+  subnet_ids=$(aws eks describe-cluster --name "$EKS_CLUSTER_NAME" --query "cluster.resourcesVpcConfig.subnetIds[]" --output text)
```

Comment:
You are trying to find the nat nodes that are in the same subnets as the EKS control plane. 

This logic will not work for all setups. There is not guaranteed to be overlap or a 1-to-1 relationship between EKS control plane subnets and the subnets that NAT nodes are running in.

Instead, use the VPC ID to filter the nat nodes.